bdai_ros2_wrappers/bdai_ros2_wrappers/tf_listener_wrapper.py

```diff
@@ -1,154 +1,120 @@
-                recent transform in the buffer.
-            wait_for_frames: If true, waits timeout amount of time for a path to exist from frame_a to frame_b in the
-                buffer.  If false, this will return immediately if a path does not exist even if timeout is not
-                None.  Note that wait_for_transform can be used to wait indefinitely for a transform to become
-                available.
-        Returns:
-            The transform frame_a_t_frame_b at the time specified.
-        Raises:
-            All the possible TransformExceptions.
-        """
-        return self._internal_lookup_a_tform_b(frame_a, frame_b, transform_time, timeout, wait_for_frames)
+        if timeout_sec != 0.0:
+            self.wait_for_a_tform_b(frame_a, frame_b, transform_time, timeout_sec)
+        return self._tf_buffer.lookup_transform(frame_a, frame_b, transform_time)
 
     def lookup_latest_timestamp(
-        self, frame_a: str, frame_b: str, timeout: Optional[float] = None, wait_for_frames: bool = False
+        self, frame_a: str, frame_b: str, timeout_sec: Optional[float] = None, wait_for_frames: bool = False
     ) -> Time:
         """
         Parameters:
             frame_a: Base frame for transform.  The transform returned will be frame_a_t_frame_b
             frame_b: Tip frame for transform.  The transform returned will be frame_a_t_frame_b
-            timeout: The time to wait for the transform to become available if the transform_time is beyond the most
-                recent transform in the buffer.
-            wait_for_frames: If true, waits timeout amount of time for a path to exist from frame_a to frame_b in the
-                buffer.  If false, this will return immediately if a path does not exist even if timeout is not
-                None.  Note that wait_for_transform can be used to wait indefinitely for a transform to become
-                available.
+            timeout_sec: The time to wait for the transform to become available if the requested time is beyond
+                the most recent transform in the buffer. If set to 0, it will not wait. If left at None, it will
+                wait indefinitely.
+            wait_for_frames: If true, it will wait for a path to exist from frame_a to frame_b in the
+                buffer. If false, lookup will fail immediately if a path between frames does not exist,
+                regardless of what timeout was set. Note that wait_for_a_tform_b can also be used to
+                wait for a transform to become available.
         Returns:
             The timestamp from the latest recorded transform frame_a_t_frame_b
         Raises:
             All the possible TransformExceptions.
         """
-        transform = self._internal_lookup_a_tform_b(frame_a, frame_b, None, timeout, wait_for_frames)
+        transform = self.lookup_a_tform_b(frame_a, frame_b, timeout_sec, wait_for_frames)
```

Comment:
Saw `self._tf_buffer.get_latest_common_time(frame_a, frame_b)` above. Is there a difference in the result from doing that verses this?

Reply:
This version can wait for frames to come into existence.

Reply:
There is a bug on line 119 right there :eyes: Fixed it and added a test (we didn't have any for this method) in https://github.com/bdaiinstitute/ros_utilities/pull/35/commits/49ee39206451b0b9b679749494eb80106982c712.